plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/sessionconfig/CodeScanSessionConfig.kt

```diff
@@ -175,39 +180,53 @@ class CodeScanSessionConfig(
 
     fun getProjectPayloadMetadata(): PayloadMetadata {
         val files = mutableSetOf<String>()
+        val traversedDirectories = mutableSetOf<VirtualFile>()
         val stack = Stack<VirtualFile>()
         var currentTotalFileSize = 0L
         var currentTotalLines = 0L
         val languageCounts = mutableMapOf<CodeWhispererProgrammingLanguage, Int>()
 
-        stack.push(projectRoot)
-        while (stack.isNotEmpty()) {
-            val current = stack.pop()
-
-            if (!current.isDirectory) {
-                if (runBlocking { !featureDevSessionContext.ignoreFile(current, this) }) {
-                    if (willExceedPayloadLimit(currentTotalFileSize, current.length)) {
-                        break
+        project.modules.forEach { module ->
+            val changeListManager = ChangeListManager.getInstance(module.project)
+            if (module.guessModuleDir() != null) {
+                stack.push(module.guessModuleDir())
+                while (stack.isNotEmpty()) {
+                    val current = stack.pop()
+
+                    if (!current.isDirectory) {
+                        // Using both featureDevSessionContext and changeListManager to ignore files.
+                        // featureDevSessionContext is used to only ignore files in gitignore in project root and uses regex pattern matching.
```

Comment:
why wouldn't we just rely on ChangeListManager? it is already gitignore aware

Reply:
its not checking gitignore files in tests. so thought it'd be prudent to keep both in. i was trying choose to leave featureDevSessionContext in or out.